lychee-lib/src/types/cache.rs

```diff
@@ -19,6 +19,28 @@ pub enum CacheStatus {
     Unsupported,
 }
 
+impl<'de> Deserialize<'de> for CacheStatus {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let status = <&str as Deserialize<'de>>::deserialize(deserializer)?;
+        match status {
+            "Excluded" => Ok(CacheStatus::Excluded),
+            "Unsupported" => Ok(CacheStatus::Unsupported),
+            other => match other.parse::<u16>() {
+                Ok(code) => match code {
+                    // classify successful status codes as cache status success
+                    200..=299 => Ok(CacheStatus::Ok(code)),
```

Comment:
Does the `--accept` option affect caching a status like 404 from an `Error` to `Ok`? If so should it be deserialized as such when present, or does it not have relevance as it's a cached value that won't be requested?

Reply:
That's a really good question — my changes do not take the `--accept` option into account. @mre what's the expected behavior when the `--accept` and the `--cache` flags are passed at the same time?

Reply:
Good question; I don't know.

I think the cache should reflect whatever the current settings were on execution to avoid confusion. The tricky part are consecutive runs with different settings. We could think about not caching those, but it would get complicated really quickly.

The alternative is to just keep the caching behavior sort of naive and see if users make any change requests.

Reply:
Makes sense to me!

This is a challenging technical problem. I've been trying to find a clean way to pass the context of the `--accept` option into the `CacheStatus` deserialization and so far have not found any easy solutions.

For now I added a comment with a `TODO` explaining that the cache status does not account for status code overrides passed through the `--accept` flag. I'm happy to revisit this in the future if needed.